libcnb-test/src/lib.rs

```diff
@@ -13,373 +13,131 @@ mod container_port_mapping;
-            app_dir: PathBuf::from(temp_app_dir.path()),
-            integration_test: self,
-        };
-
-        if output.status.success() {
-            f(integration_test_context);
-        } else {
-            panic!(
-                "pack command failed with exit-code {}!\n\npack stdout:\n{}\n\npack stderr:\n{}",
-                output
-                    .status
-                    .code()
-                    .map_or(String::from("<unknown>"), |exit_code| exit_code.to_string()),
-                integration_test_context.pack_stdout,
-                integration_test_context.pack_stderr
-            )
-        }
+    pub fn run_test<C: Borrow<TestConfig>, F: FnOnce(TestContext)>(self, config: C, f: F) {
+        self.runner
+            .run_test_internal(self.image_name.clone(), config, f);
     }
-}
-
-/// Context for a currently executing integration test.
-pub struct IntegrationTestContext<'a> {
-    /// Standard output of `pack`, interpreted as an UTF-8 string.
-    pub pack_stdout: String,
-    /// Standard error of `pack`, interpreted as an UTF-8 string.
-    pub pack_stderr: String,
-    /// The name of the image this integration test created.
-    pub image_name: String,
-    /// The directory of the app this integration test uses.
-    ///
-    /// This is a copy of the `app_dir` directory passed to [`IntegrationTest::new`] and unique to
-    /// this integration test run. It is safe to modify the directory contents inside the test.
-    pub app_dir: PathBuf,
 
-    integration_test: &'a IntegrationTest,
-}
-
-impl<'a> IntegrationTestContext<'a> {
-    /// Prepares a new container with the image from the integration test.
+    /// Starts a subsequent integration test run with inherited configuration.
     ///
-    /// This will not create nor run the container immediately. Use the returned
-    /// `PrepareContainerContext` to configure the container, then call
-    /// [`start_with_default_process`](PrepareContainerContext::start_with_default_process) on it
-    /// to actually create and start the container.
+    /// This function is the same as [`TestContext::run_test`] but automatically inherits the
+    /// configuration from the previous test run. See [`TestContext::run_test`] for details.
     ///
-    /// # Example:
-    ///
-    /// ```no_run
-    /// use libcnb_test::IntegrationTest;
-    ///
-    /// IntegrationTest::new("heroku/builder:22", "test-fixtures/empty-app").run_test(|context| {
-    ///     context.prepare_container().start_with_default_process(|container| {
-    ///         // ...
-    ///     });
-    /// });
-    /// ```
-    #[must_use]
-    pub fn prepare_container(&self) -> PrepareContainerContext {
-        PrepareContainerContext::new(self)
+    /// # Panics
+    /// - When the app could not be copied
+    /// - When this crate could not be packaged as a buildpack
+    /// - When the `pack` command unexpectedly fails
+    pub fn run_test_inherit_config<F: FnOnce(TestContext)>(self, f: F) {
```

Comment:
So just thinking out loud, but if we didn't have `run_test_inherit_config()`, then users would instead do something like this? (ie just pass `config` twice unchanged; maybe with a clone, I didn't run the below)

```rust
#[test]
fn clear_bundler_cache() {
    let config = TestConfig::new("heroku/buildpacks:20", "test-fixtures/simple-ruby-app");

    TestRunner::default().run_test(&config, |context| {
        assert_contains!(context.pack_stdout, "---> Ruby Buildpack");
        assert_contains!(context.pack_stdout, "---> Installing bundler");
        assert_contains!(context.pack_stdout, "---> Installing gems");

        context.run_test(&config, |context| {
            assert_contains!(context.pack_stdout, "---> Installing bundler");
            assert_contains!(context.pack_stdout, "---> Installing gems");
        });
    });
}
```

If so, then I wonder if we even need `run_test_inherit_config()` at all? 

Whilst it does save a small amount of boilerplate, (a) re-using the exact config only applies to a very small subset of testing scenarios (much of the time you need to change something before performing a new build, eg adjusting runtime version or whatever), (b) having a single API rather than two seems like it might be easier for people to learn/remember?

If we decided to not add it now, we could also add it later, if we find a lot of tests would benefit from it?

Anyway, I'm open to ideas here :-)